Bugfix: Fix response message when user is not the creator of the mentorship request

NenadPantelic opened the pull request:
### Description

Fixes the response message when the user is not the creator of the mentorship request.
Fixes #771.

### Type of Change:
- Code

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?
1. Tested through Swagger UI:
- Case 1:  the user didn't create the mentorship request
![image](https://user-images.githubusercontent.com/16724101/108741618-33582d80-7537-11eb-8c70-e4639f1365f8.png)
- Case 2: creator deleted the request
![image](https://user-images.githubusercontent.com/16724101/108743314-f68d3600-7538-11eb-92d6-0da4f15b5cd0.png)
- Case 3: mentorship request is not in the `pending` state
![image](https://user-images.githubusercontent.com/16724101/108744190-f0e42000-7539-11eb-9a25-6607b8a31ad8.png)
- Case 4: the user was part of the mentorship request, but not the creator
![image](https://user-images.githubusercontent.com/16724101/108744325-17a25680-753a-11eb-8578-57ff1e900c50.png)

2. Unit tests pass (all of them) -> especially observed one that tests this feature
### Checklist:

<!-- **Delete irrelevant options.** -->

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**

- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locally with my changes

NenadPantelic committed:
fix: change response message for non-creator delete request (fixes #771)
Fix response message when user is not the creator of the mentorship request, add test for that case and rename message constant

NenadPantelic commented:
@epicadk @vj-codes @devkapilbansal Can you please review this? Thanks :slightly_smiling_face:

devkapilbansal reviewed (dismissed):
So it was just reordering the conditions.

LGTM :+1: 
Thanks @NenadPantelic for this.

P.S. :- Maybe I found some similar happening in `tasks` or `task_comments` too. Can you please look into them?
Also, amend your message, it should look like:
```
fix: title
description
```

NenadPantelic commented:
> So it was just reordering the conditions.
> 
> LGTM 
> Thanks @NenadPantelic for this.
> 
> P.S. :- Maybe I found some similar happening in `tasks` or `task_comments` too. Can you please look into them?
> Also, amend your message, it should look like:
> 
> ```
> fix: title
> description
> ```

I can take a look :slightly_smiling_face:

isabelcosta reviewed (changes requested):
Thank you for working on this @NenadPantelic !
Could you add a test that would fail if you didn't do this change?
Other than this, I am ok with this PR and can accept it :)

NenadPantelic commented:
@isabelcosta There is already a test that tests this feature. But I can add test that will test the case when the mentorship relation is not in the pending state. Before this change, if the mentorship relation was not in the pending state and the removal request was sent by the non-creator of the request, it would give the `The mentorship relation is not in the pending state`, now for the same case it gives `You cannot delete a mentorship request you did not create`, so I think the only thing left is to add the test:  
when the user is not the creator and the relation is not in the pending state -> `You cannot delete a mentorship request you did not create`

NenadPantelic commented:
@isabelcosta I made the changes. 
1) added test I mentioned in the comment above
2) renamed message constant - the previous var name was not proper, it's not about just uninvolved users, but the user who didn't create the request. Non-creator of the relation request is part of the relation, so involved, but not the creator, so it's forbidden.
Please, review it again. Thanks :slightly_smiling_face:

isabelcosta reviewed (dismissed):
Amazing @NenadPantelic ! I have nothing more to point :)
Thank you for your contribution 🙌🏾

devkapilbansal reviewed (commented):
@NenadPantelic I wonder if we can break the test case you wrote in multiple test cases. :thinking:

epicadk reviewed (dismissed):
👍🏼 looks good

devkapilbansal commented on tests/mentorship_relation/test_dao_delete_request.py:
```diff
@@ -125,21 +126,87 @@ def test_dao_sender_delete_mentorship_request(self):
+            self.second_user.id, self.mentorship_relation.id
+        )
+        self.assertEqual(
+            (messages.CANT_DELETE_REQUEST_YOU_DIDNT_CREATE, HTTPStatus.FORBIDDEN),
+            result,
+        )
+        self.assertIsNotNone(
+            MentorshipRelationModel.query.filter_by(id=relation_id).first()
+        )
+
+        self.mentorship_relation.state = MentorshipRelationState.CANCELLED
+        db.session.add(self.mentorship_relation)
+        db.session.commit()
+
+        result = MentorshipRelationDAO.delete_request(
+            self.second_user.id, self.mentorship_relation.id
+        )
+        self.assertEqual(
+            (messages.CANT_DELETE_REQUEST_YOU_DIDNT_CREATE, HTTPStatus.FORBIDDEN),
+            result,
+        )
+        self.assertIsNotNone(
+            MentorshipRelationModel.query.filter_by(id=relation_id).first()
+        )
+
+        self.mentorship_relation.state = MentorshipRelationState.REJECTED
+        db.session.add(self.mentorship_relation)
+        db.session.commit()
+
+        result = MentorshipRelationDAO.delete_request(
+            self.second_user.id, self.mentorship_relation.id
+        )
+        self.assertEqual(
+            (messages.CANT_DELETE_REQUEST_YOU_DIDNT_CREATE, HTTPStatus.FORBIDDEN),
+            result,
+        )
+        self.assertIsNotNone(
+            MentorshipRelationModel.query.filter_by(id=relation_id).first()
+        )
+
```
```suggestion
       # Making a list of possible relation states excluding Pending
        relation_states = [i for i in MentorshipRelationState if i!= MentorshipRelationState.PENDING]
        
        for relation_state in relation_states:
            self.mentorship_relation.state = relation_state
            db.session.add(self.mentorship_relation)
            db.session.commit()

            result = MentorshipRelationDAO.delete_request(
                self.second_user.id, self.mentorship_relation.id
            )
            self.assertEqual(
                (messages.CANT_DELETE_REQUEST_YOU_DIDNT_CREATE, HTTPStatus.FORBIDDEN),
                result,
            )
            self.assertIsNotNone(
                MentorshipRelationModel.query.filter_by(id=relation_id).first()
            )

```

NenadPantelic replied:
I totally agree.

devkapilbansal reviewed (changes requested):
@NenadPantelic your PR looks good and changes made by you are correct.
However, the test written is not easily readable.
I would suggest using subTests for this. 
Else if we do something like the one suggested below :point_down: 

What do you think @epicadk ?

vj-codes commented:
@NenadPantelic any updates on this?

NenadPantelic committed:
Fix implementation of the unit test (user not creator, relation not in pending state)

NenadPantelic commented:
@devkapilbansal Replaced the test implementation with your implementation, thanks for the snippet, you saved me some time :100:

devkapilbansal commented on tests/mentorship_relation/test_dao_delete_request.py:
```diff
@@ -125,21 +126,49 @@ def test_dao_sender_delete_mentorship_request(self):
 
-    def test_dao_user_not_involved_tries_to_delete_mentorship_request(self):
+    def test_dao_user_not_creator_tries_to_delete_mentorship_request(self):
 
         result = MentorshipRelationDAO.delete_request(
             self.admin_user.id, self.mentorship_relation.id
         )
 
         self.assertEqual(
-            (messages.CANT_DELETE_UNINVOLVED_REQUEST, HTTPStatus.FORBIDDEN), result
+            (messages.CANT_DELETE_REQUEST_YOU_DIDNT_CREATE, HTTPStatus.FORBIDDEN),
+            result,
         )
         self.assertIsNotNone(
             MentorshipRelationModel.query.filter_by(
                 id=self.mentorship_relation.id
             ).first()
         )
 
+    def test_dao_delete_mentorship_request_user_not_creator_and_request_not_in_pending_state(
```
What's the difference in both tests?

devkapilbansal reviewed (changes requested):
@NenadPantelic can you please add one more test here, i.e. user tries to delete a request in pending state
